### ISSUE TITLE
ENH: Cleanup logging - no need to list function name in error messages

### ISSUE THREAD

**Issue by jhamman** [UW-Hydro/VIC#526]
The VIC logging macros include function and/or source file information including line numbers. This cleanup issue proposes removing all instances where the function name is listed in the logging call. This change will alleviate the chance for a function name to be incorrectly listed in the log statement, leading to confusion.  For example,

```C
if (fvar == NULL) {
    log_err("Memory allocation error in get_scatter_nc_field_float().");
}
```
would become:
```C
if (fvar == NULL) {
    log_err("Memory allocation error");
}
```

**Comment by tbohn** [comment]
Does it log the source code file name and line number?  Or do you end up with the same problem of having to hard code that in the message?

I think having the function name, or some other way to pinpoint where in the code the error occurred, is extremely helpful in debugging problems...

**Comment by jhamman** [comment]
Yes, we get the source code file name and line number without having to hard code it. I've been finding hard-coded function names in functions that they don't belong in, for example:

```C
void
myfunc()
{
   log_err("error encountered in yourfunc()";
}
```
Better to just say `log_err("error encountered") and let the compiler insert the filename and line number.

**Comment by bartnijssen** [comment]
@jhamman : We also use

```C
    fprintf(LOG_DEST, "blah blah blah");
```

in a fair number of cases. Do these need to be updated?

**Comment by jhamman** [comment]
no, these can remain as is. Nearly all of the places we use `fprintf` directly are in the print libraries which we have suggested a few refactors of, e.g. #454.

**Comment by jhamman** [comment]
closed via #581.
